request.go

```diff
@@ -279,6 +279,19 @@ func (r *Request) SetFileReader(param, fileName string, reader io.Reader) *Reque
 	return r
 }
 
+// SetMultipartData method is to set custom data using io.Reader for multipart upload.
+func (r *Request) SetMultipartData(params map[string]string, contentType string, reader io.Reader) *Request {
```

Comment:
@Asker80 Change method name to `SetMultipartField`. Your idea, it supports multiple fields.

Also update your method signature to 
```go
func (r *Request) SetMultipartField(param, fileName, contentType string, reader io.Reader) *Request {
````